docs/getting-started/01-overview/03.dependency-injection.md

```diff
@@ -1,11 +1,168 @@
+You just know they will be instantiated for you.
+
+This is a very strong pattern for testing. It's very easy to mock the dependencies if you specify your dependencies as Interfaces. Then, your mocks simply have to implement the Interface and there you go. This allows you to unit test
+your class in complete isolation.
+
+If you want to read more about Dependency Injection, here are a few links:
+* [A quick intro to Dependency Injection: what it is, and when to use it](https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/)
+* [https://blog.logrocket.com/the-perfect-architecture-flow-for-your-next-node-js-project/](https://blog.logrocket.com/the-perfect-architecture-flow-for-your-next-node-js-project/) 
+
+## Dependency Injection in NodeJS, really?
+Dependency Injection is not always popular in NodeJS. There's a big debate going on between whether you should need DI since you can import your classes (that you could be instantiated exported). In our opinion though, this still creates a tight coupling between classes. Also, testing is way harder without DI.
+
+Similar to the approach that [NestJS](https://github.com/nestjs/nest) has taken, we are going full Object Oriented Programming with Dependency Injection at its core.
+
+## TSyringe to the rescue
+
+We haven't reinvented the wheel so Pristine uses [TSyringe](https://github.com/microsoft/tsyringe), developed by Microsoft, as a Dependency Injection container. If you want to access the TSyringe container, simply get it from the kernel:
+
+```
+kernel.container
+```
+
+### Registering dependencies
+
+There are a few ways to register your dependencies. The most recommended way is to annotate your class with the `@injectable()` decorator as described in TSyringe's documentation. Here's an example:
+
+```
+@injectable()
+class DatabaseService {
+}
+
+@injectable()
+class MyService {
+    public constructor(private readonly databaseService: DatabaseService) {}    
+}
+```
+
+**Take note**, in order for Pristine and TSyringe to find your class and register it, due to the fact decorators do not get called if no classes are importing it, you must also register any services in your module class like this:
+
+```
+export const AppModule: AppModuleInterface = {
+    importServices: [
+        DatabaseService, // <-- Must be registered here to be found
+        MyService, // <-- Must be registered here to be found
+    ],
+    importModules: [
+     CoreModule,
+    ],
+    keyname: "my_namespace.app",
+}
+```
+
+#### The @tag decorator
+Even though TSyringe offers a vast range of functionalities, there's one that is missing. The ability to be injected multiple classes as an array of dependencies. Let's say you want to do something like this:
```

Comment:
I think this is false. It has the ability to do that with @injectAll, but there is no easy way to register your service. That's where @tag comes into play.